swagger.yaml

```diff
@@ -1,83 +1,192 @@
+produces: [ 'application/json' ]
+
 paths:
-  /:
+  '/shelf':
+    post:
+      description: |
+        Create a shelf.
+      parameters:
+        - name: name
+          in: formData
+          type: string
+          required: false
+          description: Name of the shelf
+      responses:
+        200:
+          description: Successful response
+    
+  '/shelf/{id}':
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      description: |
+        GET returns details about a shelf.
+      parameters:
+        - name: id
+          in: path
+          type: number
+          required: true
+          description: ID of the shelf
       responses:
         200:
-          description: Object containing information
-          schema:
-            $ref: '#/definitions/Info'
-  /sample:
+          description: Successful response
+          schema: {$ref: '#/definitions/shelf'}
+         
+    post:
+      description: |
+        POST adds an album to the shelf.
+      parameters:
+        - name: id
+          in: path
+          type: number
+          required: true
+          description: ID of the shelf
+        - name: album_id
+          in: formData
+          type: number
+          required: true
+          description: ID of the album to add
+      responses:
+        201:
+          description:
+            Success
+
+  '/shelf/{id}/albums':
     get:
-      summary: Get message
-      description: This is a sample resource that returns a short message.
+      description: |
+        GET returns a list of records for every album on the shelf.
+      parameters:
+        - name: id
+          in: path
+          description: ID of the shelf
+          required: true
+          type: number
       responses:
         200:
-          description: A short message
+          description: Successful response
           schema:
-            type: string
+            type: array
+            items: {$ref: '#/definitions/album'}
+          
+  '/album/{id}':
+    get:
+      description:
+        GET the details about an album.
+      parameters:
+        - name: id
+          in: path
+          description: ID of the album
+          required: true
+          type: number
+      responses:
+        200:
+          description: Successful response
+          schema: {$ref: '#/definitions/album'}
+          examples:
+            application/json:
+              {
+                index: 1,
+                artist: "Pink Floyd",
+                title: "Dark Side of the Moon",
+                date: "1970",
+                edition: "30th anniversary SACD",
+              }
+    put:
+      description: |
+        PUT updates an album.
+      parameters:
+        - name: id
+          in: path
+          description: ID of the album
+          required: true
+          type: number
+      responses:
+        200:
+          description: Successful response
         404:
```

Comment:
For error messages, use the error convention that we have in the skeleton: https://github.com/osu-mist/web-api-skeleton/blob/master/src/main/groovy/edu/oregonstate/mist/api/Error.groovy . It helps with implementation and keeping our error messages consistent.